src/sentry/profiles/task.py

```diff
@@ -103,26 +103,37 @@ def process_profile_task(
     sentry_sdk.set_tag("project", project.id)
     sentry_sdk.set_tag("project.slug", project.slug)
 
-    event_id = profile["event_id"] if "event_id" in profile else profile["profile_id"]
-    if "event_id" not in profile:
-        profile["event_id"] = event_id
+    profile_metadata = {
+        "organization_id": profile["organization_id"],
+        "project_id": profile["project_id"],
+    }
+
+    if "profile_id" in profile:
+        profile["event_id"] = profile["profile_id"]
+        profile_metadata["profile_id"] = profile["profile_id"]
```

Comment:
No strong feelings here but this line is redundant since it'll be added in the next if block.